android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/auth/AuthManager.java

```diff
@@ -0,0 +1,341 @@
+     *               will be cleared.
+     * @return notifies of completion or error
+     */
+    @NonNull
+    public Completable signUp(@NonNull final SignUp signUp) {
+        checkNotNull(signUp);
+        checkNotNull(signUp.getEmail());
+        checkNotNull(signUp.getPassword());
+
+        signUp.study(config.getStudyId())
+                .consent(false)
+                .status(null);
+
+        return RxUtils.toBodySingle(authenticationApi.signUp(signUp))
+                .doOnSuccess(new Action1<Message>() {
+                    @Override
+                    public void call(Message message) {
+                        authManagerDelegateProtocol.setEmail(signUp.getEmail());
+                        authManagerDelegateProtocol.setPassword(signUp.getPassword());
+                    }
+                }).doOnError(new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        authManagerDelegateProtocol.setEmail(null);
+                        authManagerDelegateProtocol.setPassword(null);
+                    }
+                }).toCompletable();
+    }
+
+    /**
+     * @param email participant's email address
+     * @return notifies of completion or error
+     */
+    @NonNull
+    public Completable resendEmailVerification(@NonNull final String email) {
+        checkNotNull(email);
+
+        return RxUtils.toBodySingle(
+                authenticationApi.resendEmailVerification(
+                        new Email()
+                                .study(config.getStudyId())
+                                .email(email)))
+                .toCompletable();
+    }
+
+    /**
+     * On success, stores participant's email, password and session. If a
+     * NotAuthenticatedException is encountered, the participant's stored password and session
+     * are cleared.
+     *
+     * @param email    participant's email address
+     * @param password participant's password
+     * @return bridge response
+     */
+    @NonNull
+    public Single<UserSessionInfo> signIn(@NonNull final String email, @NonNull final String
+            password) {
+        checkNotNull(email);
+        checkNotNull(password);
+
+        SignIn signIn = new SignIn()
+                .study(config.getStudyId())
+                .email(email)
+                .password(password);
+
+        return RxUtils.toBodySingle(
+                authenticationApi.signIn(signIn))
+                .doOnSuccess(new Action1<UserSessionInfo>() {
+                    @Override
+                    public void call(UserSessionInfo userSessionInfo) {
+                        authManagerDelegateProtocol.setEmail(email);
+                        authManagerDelegateProtocol.setPassword(password);
+                        authManagerDelegateProtocol.setUserSessionInfo(userSessionInfo);
+                    }
+                }).doOnError(new Action1<Throwable>() {
```

Comment:
Should this be the place where we intercept the ConsentRequiredException?  Or will that be done by the AuthManagerDelegate, which I expect to be the BridgeDataProvider